whelktool/scripts/cleanups/2020/04/lxl-1625-notatedMusicType.groovy

```diff
@@ -0,0 +1,49 @@
+PrintWriter scheduledForUpdating = getReportWriter("scheduled-updates")
+PrintWriter failedUpdating = getReportWriter("failed-updates")
+
+String where = "data#>>'{@graph,1,@type}' = 'NotatedMusicInstance' and data#>>'{@graph,2,@type}' = 'NotatedMusic'"
+
+selectBySqlWhere(where) { data ->
+    def (record, instance, work) = data.graph
+    boolean changed = false
+
+    if (instance["marc:mediaTerm"] && instance["marc:mediaTerm"] instanceof String) {
+
+        if (instance["marc:mediaTerm"].contains("tryck") ||
+                instance["marc:mediaTerm"].contains("druck") ||
+                instance["marc:mediaTerm"].contains("Noten") ||
+                instance["marc:mediaTerm"].contains("tryk") ) {
+            instance["@type"] = "Print"
+            changed = true
+        }
+
+        if (instance["marc:mediaTerm"].contains("Elektronisk")) {
+            instance["@type"] = "Electronic"
+            changed = true
+        }
+
+        if (instance["marc:mediaTerm"].contains("handskrift")) {
+            instance["@type"] = "Manuscript"
+            changed = true
+        }
+    }
+
+    if (!changed) {
+        instance["@type"] = "Instance"
+        changed = true
+    }
+
+    if (instance["hasType"] != null) {
```

Comment:
Two questions:
- Should `hasType` instead be `hasPart`? :)
- Do we know that `'{@graph,1,hasPart,@type}'` always exists in the sql query selection above? `(String where = "data#>>'{@graph,1,@type}' = 'NotatedMusicInstance' and data#>>'{@graph,2,@type}' = 'NotatedMusic'")`. We might need to adjust the sql query?

Reply:
1. Yes it should be, sorry typo in my spec. 🙈 
2. Indeed you are right again, we can not presume to know in what order they exist.

Reply:
I'm embarrased :) well spotted!